exist-core/src/main/java/org/exist/source/SourceFactory.java

```diff
@@ -152,7 +152,7 @@ private static String firstPathSegment(final String path) {
     }
 
     private static Source getSource_fromClasspath(final String contextPath, final String location) throws IOException {
-        if (location.startsWith(ClassLoaderSource.PROTOCOL)) {
+        if (contextPath == null || location.startsWith(ClassLoaderSource.PROTOCOL)) {
```

Comment:
It appears to me as if this case should be handled further up in `getSource` or a separate method `getSource_fromLocation`

Reply:
`if (location.startsWith(ClassLoaderSource.PROTOCOL))` is duplicated in both methods now for now added value.

Reply:
What do you think?

Reply:
This is now fixed in a subsequent commit.